lib/pages/bootstrap/tom_bootstrap_dialog.dart

```diff
@@ -1,112 +1,184 @@
-    this.recoveryWords,
-    this.wipe = false,
-    this.wipeRecovery = false,
     required this.client,
   });
 
   Future<bool?> show() => TwakeDialog.showDialogFullScreen(
         builder: () => this,
+        barrierColor: TomBootstrapDialogStyle.barrierColor,
       );
 
   @override
   TomBootstrapDialogState createState() => TomBootstrapDialogState();
 }
 
-class TomBootstrapDialogState extends State<TomBootstrapDialog> {
+class TomBootstrapDialogState extends State<TomBootstrapDialog>
+    with TickerProviderStateMixin {
   final _saveRecoveryWordsInteractor = getIt.get<SaveRecoveryWordsInteractor>();
+
+  final _getRecoveryWordsInteractor = getIt.get<GetRecoveryWordsInteractor>();
+
   final _deleteRecoveryWordsInteractor =
       getIt.get<DeleteRecoveryWordsInteractor>();
   Bootstrap? bootstrap;
 
-  String? titleText;
-  Widget? body;
-  final buttons = <AdaptiveFlatButton>[];
-
   UploadRecoveryKeyState _uploadRecoveryKeyState =
-      UploadRecoveryKeyState.initial;
+      UploadRecoveryKeyState.dataLoading;
 
-  bool? _wipe;
+  bool _wipe = false;
+  RecoveryWords? _recoveryWords;
 
   @override
   void initState() {
     super.initState();
-    _createBootstrap(widget.wipe);
+    _createBootstrap();
   }
 
-  void _createBootstrap(bool wipe) async {
-    _wipe = wipe;
-    titleText = null;
-    _uploadRecoveryKeyState = _initializeRecoveryKeyState();
-    WidgetsBinding.instance.addPostFrameCallback((_) {
-      bootstrap =
-          widget.client.encryption!.bootstrap(onUpdate: (_) => setState(() {}));
+  void _createBootstrap() async {
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      await _loadingData();
     });
   }
 
-  UploadRecoveryKeyState _initializeRecoveryKeyState() {
-    if (widget.wipeRecovery) {
-      return UploadRecoveryKeyState.wipeRecovery;
+  Future<void> setupAdditionalDioCacheOption(String userId) async {
+    Logs().d('TomBootstrapDialog::setupAdditionalDioCacheOption: $userId');
+    DioCacheInterceptorForClient(userId).setup(getIt);
+  }
+
+  Future<RecoveryWords?> _getRecoveryWords() async {
+    try {
+      return await _getRecoveryWordsInteractor.execute().then(
+            (either) => either.fold(
+              (failure) => null,
+              (success) => success.words,
+            ),
+          );
+    } catch (_) {
+      rethrow;
     }
+  }
 
-    if (widget.recoveryWords != null) {
-      return UploadRecoveryKeyState.useExisting;
+  Future<void> _loadingData() async {
+    _uploadRecoveryKeyState = UploadRecoveryKeyState.dataLoading;
+    await widget.client.roomsLoading;
+    await widget.client.accountDataLoading;
+    if (widget.client.userID != null) {
+      await setupAdditionalDioCacheOption(widget.client.userID!);
     }
+    setState(() {
+      _uploadRecoveryKeyState = UploadRecoveryKeyState.checkingRecoveryWork;
+    });
+    await _getRecoveryKeyState();
+    bootstrap =
+        widget.client.encryption!.bootstrap(onUpdate: (_) => setState(() {}));
```

Comment:
Causes null pointer exception on my computer. Please avoid using exclamation mark without checking if the value is null or not.

![Capture d’écran du 2024-06-06 10-32-00](https://github.com/linagora/twake-on-matrix/assets/31937920/6521537e-fd3a-41bd-a54a-d97b25b059d6)

Reply:
I fixed it on last fixup commit, pls verify it again